delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/directives/ResponseToSse.scala

```diff
@@ -0,0 +1,81 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.directives
+
+import akka.http.scaladsl.marshalling.sse.EventStreamMarshalling._
+import akka.http.scaladsl.model.StatusCodes.OK
+import akka.http.scaladsl.model.sse.ServerSentEvent
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import akka.persistence.query.{NoOffset, Sequence, TimeBasedUUID}
+import akka.stream.scaladsl.Source
+import cats.syntax.functor._
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.rdf.utils.JsonKeyOrdering
+import ch.epfl.bluebrain.nexus.delta.sdk.JsonValue
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.DeltaDirectives.emit
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.Response.{Complete, Reject}
+import ch.epfl.bluebrain.nexus.delta.sdk.marshalling.HttpResponseFields
+import ch.epfl.bluebrain.nexus.delta.sdk.marshalling.RdfMarshalling._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.Envelope
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import fs2.Stream
+import io.circe.Encoder
+import monix.bio.{IO, Task, UIO}
+import monix.execution.Scheduler
+import streamz.converter._
+
+sealed trait ResponseToSse {
+  def apply(): Route
+}
+
+object ResponseToSse {
+
+  private def apply[E: JsonLdEncoder](
+      io: IO[Response[E], Stream[Task, Envelope[JsonValue]]]
+  )(implicit s: Scheduler, jo: JsonKeyOrdering, cr: RemoteContextResolution): ResponseToSse =
+    new ResponseToSse {
+
+      private def toSse(envelope: Envelope[JsonValue]) = {
+        val json       = envelope.event.encoder(envelope.event.value)
+        val id: String = envelope.offset match {
+          case TimeBasedUUID(value) => value.toString
+          case Sequence(value)      => value.toString
+          case NoOffset             => "-1"
```

Comment:
I think it would be better to have this:

```
val idOpt: Option[String] = envelope.offset match {
  case TimeBasedUUID(value) => Some(value.toString)
  case Sequence(value)      => Some(value.toString)
  case NoOffset             => None
}
ServerSentEvent(defaultPrinter.print(json.sort), Some(envelope.eventType), idOpt)
```